### PR TITLE
Refactor Home page to Vue3

### PR DESCRIPTION
**Pull request by guergana**
Bug: [T353869](https://phabricator.wikimedia.org/T353869)

### CODE REVIEW

File: resources/js/Pages/Home.vue [hunk 1]
```diff
@@ -121,62 +116,41 @@
...
+    const textareaInputValue: Ref<string> = ref(store.lastSearchedIds);
+    const faqDialog = ref(false);
+
+    function send(): void{
+        (textarea as InstanceType<typeof ItemIdSearchTextarea>).value.validate();
+
+        if((textarea as InstanceType<typeof ItemIdSearchTextarea>).value.validationError) {
+            return;
         }
+        const store = useStore();
```
Comment (itamargiv) [inline resources/js/Pages/Home.vue:131]: We already have store in scope, we don't need to use it again.

**Review by itamargiv — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments.

File: resources/js/Pages/Home.vue [hunk 2]
```diff
@@ -121,62 +116,41 @@
...
+        return errors && Object.keys(errors).length > 0;
+    });
+
+    const unexpectedError = computed<string>(() => {
+        const flashMessages = page.props.value.flash as FlashMessages;
+        return (flashMessages.errors && flashMessages.errors.unexpected);
+    });
+
+    const loading = computed<boolean>(() => {
+        return (store.loading);
```
[the remark this answers is not in the document]
Comment (guergana) [reply resources/js/Pages/Home.vue:153]: Why is having the computed before the functions more consistent?

File: resources/js/Pages/Home.vue (hunk 1 above)
Comment (guergana) [reply resources/js/Pages/Home.vue:131]: oops, yes, that was a copy/paste bloop. good catch. thx

File: resources/js/Pages/Home.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (itamargiv) [reply resources/js/Pages/Home.vue]: yep

**Review by itamargiv — APPROVED** [review]
Looks like it works well, thank you!

**Pull request merged**